Remove play button/Start timer only when a game is started/Refactor g…

joshpham97 opened the pull request:
…ame difficulty selection

joshpham97 committed:
Remove play button/Start timer only when a game is started/Refactor game difficulty selection

joshpham97 committed:
Remove play button/Start timer only when a game is started/Refactor game difficulty selection

joshpham97 committed:
UI adjustments and refactor

Added a UI for when the application first run (including moving frame initialization outside of board_ui())
Display the timer only when a game is in progress
Adjust borders of UI elements

joshpham97 committed:
UI adjustments and refactor

Added a UI for when the application first run (including moving frame initialization outside of board_ui())
Display the timer only when a game is in progress
Adjust borders of UI elements

changlmasaki committed:
Add more notations for the test

joshpham97 commented:
I will remove one of them later, since we do not need both.

binarytrails commented:
> I will remove one of them later, since we do not need both.

@joshpham97 got it! perhaps, a TODO  comment would be useful cause I couldn't understand

changlmasaki committed:
fix indentation

binarytrails committed:
Merge pull request #49 from binarytrails/Test/Chang-2

Add more notations for the test

joshpham97 commented:
Oh my bad I forgot to put it

changlmasaki committed:
add TestGameController

Tiffany910 committed:
Ia

binarytrails committed:
Merge pull request #51 from binarytrails/diary_TA

I added  meetings times for the previous weeks

changlmasaki committed:
Update TestGameController.java

changlmasaki committed:
Update TestGameController.java

binarytrails committed:
Merge pull request #50 from binarytrails/Test/Chang-2

add TestGameController

joshpham97 committed:
Refactor code

Split all components in Models, Views, Controller if not done.
Created a hierarchy between the controllers.
AppController is the root controller --> Contrains the 3 remaining controllers: Chrono, Board and Button Menu. It also handles the main functionality of the program (save, load, pause, etc. where multiple components interactions are required).
AppView is the root view --> Contains the 3 remaining views as above.
The Chrono Model is added to models.
Any interactions between subviews and the root controller must go throught subcontroller (eg: if the button menu view wants to call a method in the root controller it must go through the button menu controller first, then the button menu controller will call the root controller).
Divided the code into folders. Anything that is not related to MVC or database goes into core.
Updated some test cases.

binarytrails commented:
@joshpham97 
to merge with ``` git rebase -i master```, I use http://meldmerge.org/ you can do ```git mergetool --tool meld``` and keep going with ```git merge --continue``` until it is all good and then ```git push -f```

joshpham97 committed:
Refactor the code and fixed merge conglicts and commented out TestGameController

joshpham97 committed:
Fixed merge conflicts

binarytrails commented on diaries/diary_TiffanyAH KING.tex:
```diff
@@ -172,7 +172,46 @@ \section{Iteration 1}
 
 
-%\section{Iteration 2}
+\section{Iteration 2}
```
that is strange it is already on master see: https://github.com/binarytrails/kakuro/commit/a26843f41e33c9e99ada0b5c32e45b9dff65bbed

binarytrails commented on tests/kakuro/TestPlayerDao.java:
```diff
@@ -1,47 +1,40 @@
 	@Test
-	public void testRegisterAlreadyRegistered() {
-		
+	public void testRegisterAlreadyRegistered(){
```
same for tests already on master, there is something odd here

binarytrails commented:
in any cases the tests gotta pass ! aim for that

joshpham97 committed:
Fixed test bug and renamed App Controller to Game Controller for consistency

joshpham97 committed:
Fixed more test cases
